src/storage/index/IndexPolicyMaker.cpp

```diff
@@ -0,0 +1,149 @@
+                 */
+                hintNum++;
+                auto v = std::get<1>(*it);
+                hasStr = (v.which() == VAR_STR);
+                prefix_.append(NebulaKeyUtils::encodeVariant(v));
+            } else {
+                break;
+            }
+        } else {
+            break;
+        }
+    }
+    if (hintNum == operatorList_.size() && !hasStr) {
+        isIndexScan_ = false;
+    }
+}
+
+cpp2::ErrorCode IndexPolicyMaker::traversalExpression(const Expression *expr) {
+    cpp2::ErrorCode code = cpp2::ErrorCode::SUCCEEDED;
+    if (!optimizedPolicy_) {
+        return code;
+    }
+    Getters getters;
+    switch (expr->kind()) {
+        case nebula::Expression::kLogical : {
+            auto* lExpr = dynamic_cast<const LogicalExpression*>(expr);
+            if (lExpr->op() == LogicalExpression::Operator::XOR) {
+                return cpp2::ErrorCode::E_INVALID_FILTER;
+            } else if (lExpr->op() == LogicalExpression::Operator::OR) {
+                optimizedPolicy_ = false;
+                return code;
+            }
+            auto* left = lExpr->left();
+            traversalExpression(left);
+            auto* right = lExpr->right();
+            traversalExpression(right);
+            break;
+        }
+        case nebula::Expression::kRelational : {
+            std::string prop;
+            VariantType v;
+            auto* rExpr = dynamic_cast<const RelationalExpression*>(expr);
+            auto* left = rExpr->left();
+            auto* right = rExpr->right();
+            if (left->kind() == nebula::Expression::kAliasProp
+                && right->kind() == nebula::Expression::kPrimary) {
+                auto* aExpr = dynamic_cast<const AliasPropertyExpression*>(left);
+                prop = *aExpr->prop();
+                auto* pExpr = dynamic_cast<const PrimaryExpression*>(right);
+                auto value = pExpr->eval(getters);
+                if (!value.ok()) {
+                    return cpp2::ErrorCode::E_INVALID_FILTER;
+                }
+                v = value.value();
+            } else if (left->kind() == nebula::Expression::kPrimary
+                       && right->kind() == nebula::Expression::kAliasProp) {
+                auto* pExpr = dynamic_cast<const PrimaryExpression*>(left);
+                auto value = pExpr->eval(getters);
+                if (!value.ok()) {
+                    return cpp2::ErrorCode::E_INVALID_FILTER;
+                }
+                v = value.value();
+                auto* aExpr = dynamic_cast<const AliasPropertyExpression*>(right);
+                prop = *aExpr->prop();
+            } else {
+                traversalExpression(left);
+                traversalExpression(right);
+                break;
+            }
+            operatorList_.emplace_back(std::make_tuple(std::move(prop), std::move(v), rExpr->op()));
+            break;
+        }
+        case nebula::Expression::kFunctionCall : {
```

Comment:
`kUUID` maybe same with `kFunctionCall `

Reply:
Good point, In the future, we need add the logic when uuid supprot where clause in the future.